### PR TITLE
Add b.p.fuelCladLocked

### PR DESCRIPTION
**Pull request by albeanth**
## Description
This PR adds a block parameter to indicate if the fuel and clad are physically locked. This is a common interaction of interest in SFRs.

Also cleaning up `_isFuelLocked` in the axialExpansionChanger to leverage `ArmiObject::getComponent`. 

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.
    - Its use is reserved for separate ARMI plugins.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

**Comment by albeanth** [comment]
@sammiller11235

File: armi/reactor/converters/axialExpansionChanger.py [hunk 1]
```diff
@@ -818,14 +818,8 @@ def _isFuelLocked(self, b):
...
-        c = b.getChildrenWithFlags(Flags.FUEL)
-        if len(c) == 0:  # pylint: disable=no-else-raise
-            raise RuntimeError("No fuel component within {0}!".format(b))
-        elif len(c) > 1:
-            raise RuntimeError(
-                "Cannot have more than one fuel component within {0}!".format(b)
-            )
-        self._componentDeterminesBlockHeight[c[0]] = True
+        c = b.getComponent(Flags.FUEL)
+        self._componentDeterminesBlockHeight[c] = True
```
Comment (jakehader) [inline armi/reactor/converters/axialExpansionChanger.py:822]: How does this prevent against issues that the previous implementation checked for? `getComponent` will return a `None` if there is no `FUEL` here. Will this work still or should we still check that?

File: armi/physics/fuelPerformance/parameters.py [hunk 2]
```diff
@@ -27,6 +27,13 @@ def _getFuelPerformanceBlockParams():
     pDefs = parameters.ParameterDefinitionCollection()
     with pDefs.createBuilder(default=0.0, location=ParamLocation.AVERAGE) as pb:
 
+        pb.defParam(
+            "fuelCladLocked",
+            units=bool,
```
Comment (jakehader) [inline armi/physics/fuelPerformance/parameters.py:32]: Units is not a boolean, right?

Comment (albeanth) [reply armi/physics/fuelPerformance/parameters.py:32]: Yeah, the boolean isn't the units. I was thinking that maybe we could use it to enfore type, but now that you comment on it, that doesn't make sense. I'll pop it out.

File: armi/reactor/converters/axialExpansionChanger.py (hunk 1 above)
Comment (albeanth) [reply armi/reactor/converters/axialExpansionChanger.py:822]: It won't. Good catch. I was going too fast and read the `runLog.warning` as a RuntimeError.
https://github.com/terrapower/armi/blob/e47abd2690e3833bf8e3ed02225df2ddea6bc017/armi/reactor/composites.py#L2477-L2483

I'll put that specific check back.

File: armi/physics/fuelPerformance/parameters.py (hunk 2 above)
Comment (sammiller11235) [reply armi/physics/fuelPerformance/parameters.py:32]: the type gets inferred from the default

Comment (albeanth) [reply armi/physics/fuelPerformance/parameters.py]: Addressed https://github.com/terrapower/armi/pull/1038/commits/67dee5406cd9fa7da82cfc1287a7b636500956d9

**Comment by albeanth** [comment]
@john-science I think it's safe to say coverage drop is coverage spray here.

File: armi/physics/fuelPerformance/parameters.py [hunk 3]
```diff
@@ -27,6 +27,13 @@ def _getFuelPerformanceBlockParams():
     pDefs = parameters.ParameterDefinitionCollection()
     with pDefs.createBuilder(default=0.0, location=ParamLocation.AVERAGE) as pb:
 
+        pb.defParam(
+            "fuelCladLocked",
+            units="",
+            default=False,
+            description="Boolean to indicate if the fuel is locked with the clad.",
```
Comment (jakehader) [inline armi/physics/fuelPerformance/parameters.py:34]: Can you add to this description a bit:

`Boolean to indicate if the fuel is locked with the clad. This is used to determine the expansion constraints for the fuel during thermal and/or burn-up expansion of the fuel and cladding materials.`

File: armi/reactor/converters/tests/test_axialExpansionChanger.py [hunk 4]
```diff
@@ -662,34 +662,19 @@ def test_determineTargetComponentRuntimeErrorSecond(self):
             self.assertEqual(the_exception.error_code, 3)
 
     def test_isFuelLocked(self):
```
Comment (jakehader) [inline armi/reactor/converters/tests/test_axialExpansionChanger.py:664]: Please add a docstring to this test to explain what it does and how it works.

**Review by jakehader — CHANGES_REQUESTED** [review]
Comments

File: armi/physics/fuelPerformance/parameters.py (hunk 3 above)
Comment (albeanth) [reply armi/physics/fuelPerformance/parameters.py]: https://github.com/terrapower/armi/pull/1038/commits/a04e0b8aede3e340699ca9b8d45159f369607af1

File: armi/reactor/converters/tests/test_axialExpansionChanger.py (hunk 4 above)
Comment (albeanth) [reply armi/reactor/converters/tests/test_axialExpansionChanger.py:664]: https://github.com/terrapower/armi/pull/1038/commits/b940f8bb322b970b0caeafca7da4c56e809f8a49

Comment (jakehader) [reply armi/reactor/converters/tests/test_axialExpansionChanger.py:664]: Extend to say something like:

```
"""
Ensures that the RuntimeError statement in ExpansionData::_isFuelLocked is raised appropriately.

Notes
------
This is implemented by creating a fuel block that contains no fuel component. 
"""
```

File: armi/reactor/converters/axialExpansionChanger.py [hunk 5]
```diff
@@ -818,14 +818,10 @@ def _isFuelLocked(self, b):
...
-            raise RuntimeError("No fuel component within {0}!".format(b))
-        elif len(c) > 1:
-            raise RuntimeError(
-                "Cannot have more than one fuel component within {0}!".format(b)
-            )
-        self._componentDeterminesBlockHeight[c[0]] = True
+        c = b.getComponent(Flags.FUEL)
+        if c is None:
+            raise RuntimeError(f"No fuel component within {b}!")
+        self._componentDeterminesBlockHeight[c] = True
```
Comment (jakehader) [inline armi/reactor/converters/axialExpansionChanger.py:824]: Actually, this is confusing to me, don't you need to use the `fuelCladLocked` parameter too?

**Review by jakehader — CHANGES_REQUESTED** [review]
Tag, you're it.

File: armi/reactor/converters/tests/test_axialExpansionChanger.py (hunk 4 above)
Comment (albeanth) [reply armi/reactor/converters/tests/test_axialExpansionChanger.py:664]: https://github.com/terrapower/armi/pull/1038/commits/a4a561f26b84d2ae7dc661017a3168636bdb305f

File: armi/reactor/converters/axialExpansionChanger.py (hunk 5 above)
Comment (jakehader) [reply armi/reactor/converters/axialExpansionChanger.py:824]: Discussed offline. This is good for as a base class to build on.

**Review by jakehader — APPROVED** [review]

**Review by john-science — APPROVED** [review]
Sure, we discussed the reason for this.  It's a good parameter to have in ARMI. 

Just ONE LINE of code coverage missing. So close!

**Pull request merged**